src/common/mod.rs

```diff
@@ -0,0 +1,38 @@
+//! Common helpers for `displayz`
+
+/// Height and Width of a Display (`i32`)
+pub struct Resolution(i32, i32);
+
+/// X/Y positions of a display.
+pub struct Position(i32, i32);
+
+/// `Vec` type of the `Display` struct, exposed on a platform-dependent basis.
+pub type Displays = Vec<crate::Display>;
+
+/// `Vec` type of the `Resolution` type, generally exposing a collection of available resolutions.
+pub type Resolutions = Vec<Resolution>;
+
+/// `DisplayOutput` defines the Trait specification of a platform's `Display`.
+/// A `Display` may contain:
+/// - An EDID.
+/// - State, including if the `Display` is active, primary, or has a fault.
+/// - Other helper methods, returning data stored in-memory state.
+pub trait DisplayOutput {
+    /// Returns a boolean result, if the `Display` is the 'primary display' or not.
+    fn is_primary(&self) -> bool;
+    /// Returns a boolean result, if the `Display` is currently active or not.
+    fn is_active(&self) -> bool;
+    /// Returns a boolean result, if the `Display` is currently focused or not.
+    /// This is not available on all platforms, in the event it's not, it will *always* return `false`.
+    fn is_focused(&self) -> bool {
```

Comment:
This could be implemented in a `DisplayOutputXExt` trait (long name, I know!), where 'X' is the platform instead.

Reply:
Removed `is_focused` for now. On Sway, there's a `focused` attribute when you run `swaymsg -t get_outputs`, and I think, if that's a Wayland-specific attribute, I could make a `DisplayOutputWaylandExt` trait with that function.

Reply:
a19d993.